### PR TITLE
fix Connected to undefined message

### PR DESCRIPTION
**Pull request by fauna-chase**
ENG-5630
This will display the endpoint and database being connected to when ever a CLI command is executed.

```
p4 » local-shell schema diff
Connected to endpoint: cli_test-us database: cli_bugbash
No schema differences
```

```
p4 » local-shell shell
Connected to endpoint: cli_test-us database: cli_bugbash
Type Ctrl+D or .exit to exit the shell
>
```

### CODE REVIEW

**Comment by fauna-chase** [comment]
copying my jira comment here:

the previous impl was just printing the url it was connecting to.  I updated the current impl to do that, but do we think printing that url has any value?  for customers it will always be https://db.fauna.com or localhost feels like we could just strip it out entirely.

One thing that does seem valuable is printing the database that is being connected to.  Should we update it to do that?

**Comment by macmv** [comment]
I like the idea of printing the current endpoint and database that its connected to. URL seems nice to have too, if we can fit all 3 of those in that one-liner.

File: src/lib/fauna-command.js [hunk 1]
```diff
@@ -85,9 +84,7 @@ class FaunaCommand extends Command {
   mapConnectionError({ err, connectionOptions }) {
     if (err instanceof errors.Unauthorized) {
       return this.error(
-        `Could not Connect to ${stringifyEndpoint(
-          connectionOptions
-        )} Unauthorized Secret`
+        `Could not Connect to ${connectionOptions.url} Unauthorized Secret`
```
Comment (macmv) [inline src/lib/fauna-command.js:87]:
```suggestion
        `Could not connect to ${connectionOptions.url}: unauthorized secret`
```

Comment (fauna-chase) [reply src/lib/fauna-command.js:87]: hmm yeah I'm not sure that we want to do that, my bias would be to just leave it out

File: src/lib/config/root-config.ts [hunk 2]
```diff
@@ -101,14 +101,16 @@ export class RootConfig {
  * - An optional GraphQL port. This is the port to use for GraphQL requests. It defaults to `443`.
  */
 export class Endpoint {
+  name?: string;
```
Comment (macmv) [inline src/lib/config/root-config.ts:104]:
```suggestion
  name: string;
```

File: src/lib/config/root-config.ts [hunk 3]
```diff
@@ -101,14 +101,16 @@ export class RootConfig {
...
   secret: string;
   url: string;
 
   graphqlHost: string;
   graphqlPort: number;
 
-  static fromConfig(config: Config) {
+  static fromConfig(name: string, config: Config) {
     return new Endpoint({
+      name: name,
```
Comment (macmv) [inline src/lib/config/root-config.ts:113]: style:

```suggestion
      name,
```

File: src/lib/config/root-config.ts [hunk 4]
```diff
@@ -118,41 +120,40 @@ export class Endpoint {
   }
 
   constructor(opts: {
+    name?: string;
```
Comment (macmv) [inline src/lib/config/root-config.ts:123]:
```suggestion
    name: string;
```

File: test/commands/endpoint.test.ts [hunk 5]
```diff
@@ -114,6 +117,7 @@ describe("endpoint:add", () => {
           foobar: {
             url: "http://foo.baz",
             secret: "fn1234",
+            name: undefined,
```
Comment (macmv) [inline test/commands/endpoint.test.ts:120]: can you update `name` to always be defined? it seems strange to only set it when reading the config, but not when writing it.

**Review by macmv — COMMENTED** [review]
looks good to me, except for name being optional. IMO it should always be set.

**Comment by fauna-chase** [comment]
> looks good to me, except for name being optional. IMO it should always be set.

if you provide a secret as a flag we create an endpoint without a name.

**Comment by macmv** [comment]
oh yup nevermind

**Review by macmv — APPROVED** [review]

**Pull request merged**